unistd/sys.c

```diff
@@ -60,6 +61,23 @@ static int shebang(const char *path)
 	return fd;
 }
 
+static int checkelf(const char *path)
```

Comment:
Wouldn't this screw up scripts executions (#!/bin/bash)?

Reply:
If we execute a script this comes into play:
```c
if ((fd = shebang(file)) >= 0) {
(...)
	while (*interp == ' ') ++interp;
	end = interp;
	while (*end && !isspace(*end)) ++end;
	*end = 0;
(...)
	file = interp;
	argv = sb_args;
}
```

which substitutes the `file` path with interpreter path (for `#!/bin/bash` -> `/bin/bash`) which is later resolved to interpreter binary file which is good.

Also I checked against `psh` scripts adding a file called `abc` which contains only:
> !/bin/psh

and it produces this output:
> psh: /abc is not a psh script

which means that passing to script works well. After all, syscall `exec()` is **only** for binary files. Only arguments change so user sees it as _executing a script_